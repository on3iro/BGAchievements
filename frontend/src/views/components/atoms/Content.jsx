// @flow

import styled from 'styled-components'

const Content = styled.div`
  flex-grow: 1;

  padding: 56px 24px 24px;
  min-height: calc(100% - 56px);
`

export {
  Content as default
}
