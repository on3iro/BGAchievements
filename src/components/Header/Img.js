/**
  * Renders a styled img
  *
  * @namespace Header.Img
  * @memberOf Header
  */

import styled from 'styled-components'

import NormalImg from 'components/Img'

const Img = styled(NormalImg)`
  width: 70px;
  display: block;
`

export default Img
