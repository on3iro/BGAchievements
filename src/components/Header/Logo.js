/**
  * Renders a reformatted anchor
  *
  * @namespace Header.A
  * @memberOf Header
  */

import styled from 'styled-components'
import { Link } from 'react-router-dom'

const Logo = styled(Link)`
  display: inline-block;
  margin: 10px 80px;
  outline: 0;
`

export default Logo
