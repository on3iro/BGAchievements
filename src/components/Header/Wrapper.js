/**
  * Styled container for Header
  *
  *
  * @namespace Header.Wrapper
  * @memberOf Header
  */

import styled from 'styled-components'

const Wrapper = styled.header`
  position: fixed;
  top: 0px;
  margin-bottom: 0;

  min-height: 64px;
  height: 64px;
  width: 100%;

  background-color: ${props => props.theme.mainBG};
  color: ${props => props.theme.noneWhite};

  @media (max-width: 900px) {
    position: static;

    height: auto;
  }
`

export default Wrapper
