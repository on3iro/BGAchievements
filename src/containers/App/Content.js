/**
  * Styled Wrapper for site content
  *
  * @namespace App.Content
  * @memberOf App
  */

import styled from 'styled-components'

const Content = styled.main`
  margin: 10px auto 60px auto;
  padding: 0;
  width: 100%;
  height: 60%;

  @media (max-width: 900px) {
    padding: 20px;
    height: 100%;
  }
`

export default Content
