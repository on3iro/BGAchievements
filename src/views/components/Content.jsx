// @flow

import styled from 'styled-components'

const Content = styled.div`
  flex-grow: 1;

  padding: 20px;
  max-width: 60%;
  min-height: calc(100% - 40px);
`

export {
  Content as default
}
