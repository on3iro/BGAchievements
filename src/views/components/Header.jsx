// @flow

import React from 'react'
import styled from 'styled-components'

import Logo from './Logo'
import NavMain from './NavMain'

const Wrapper = styled.div`
  display: flex;

  flex-direction: row;
  justify-content: flex-start;

  width: 100%;
  height: 60px;
  min-height: 60px;

  background: ${props => props.theme.primary};
  color: ${props => props.theme.white};
`

const Header = () => (
  <Wrapper>
    <Logo>BGAchievements</Logo>
    <NavMain />
  </Wrapper>
)

export {
  Header as default
}
