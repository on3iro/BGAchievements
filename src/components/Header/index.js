/**
  * Renders a Header with a Logo and a navigation
  *
  * @namespace Header
  */

import React from 'react'
import Grid from 'grid-styled'

import Navigation from 'components/Navigation'
import UserNav from 'components/UserNav'

import Logo from './Logo'
import Img from './Img'
import Wrapper from './Wrapper'

export const Header = props => {
  return (
    <Wrapper>
      <Grid sm={1 / 1} md={1 / 6}>
        <Logo to='/'>
          <Img src='' alt='placeholder' />
        </Logo>
      </Grid>
      <Grid sm={1 / 1} md={4 / 6}>
        <Navigation {...props} />
      </Grid>
      <Grid sm={1 / 1} md={1 / 6}>
        <UserNav {...props} />
      </Grid>
    </Wrapper>
  )
}

export default Header
