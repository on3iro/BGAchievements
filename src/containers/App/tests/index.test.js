import React from 'react'
import { shallow } from 'enzyme'

import { App, mapStateToProps } from '../index'
import Content from '../Content'
import Header from 'components/Header'
import Footer from 'components/Footer'

describe('App', () => {
  it('Should render a <Header />', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find(Header).length).toEqual(1)
  })

  it('Should render a <Footer />', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find(Footer).length).toEqual(1)
  })

  describe('react-redux', () => {
    it('should map state to props', () => {
      expect(mapStateToProps({
        auth: { user: { isLoggedIn: true }},
        router: { location: { pathname: '/login' }}
      })).toMatchSnapshot()
    })
  })
})
