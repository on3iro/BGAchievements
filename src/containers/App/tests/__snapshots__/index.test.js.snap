exports[`App react-redux should map state to props 1`] = `
Object {
  "isLoggedIn": true,
  "routerLocation": Object {
    "pathname": "/login",
  },
}
`
